### ISSUE TITLE
Add support for venvs.

### ISSUE THREAD

**Issue by BeastLe9enD** [PyO3/pyo3#3284]
I have a python venv and in my opinion, there should be a function to load the venv into the pyo3 runtime. Something like
```Rust
pyo3::set_venv("/path/to/your/venv");
```
would make things very easy. Or is there a simple way how this is possible that I'm missing?

**Comment by adamreichold** [comment]
Usually, activating the venv before building your project should be sufficient as we normally use the first `python3` binary found on the path to determine the interpreter configuration.

**Comment by mobile-bungalow** [comment]
those of us on MacOs are still waiting on #1741.  additionally, hot swapping venvs at runtime has more surface area than just updating the path as suggested in the linked issue. You also have to remove the sites-packages paths from preexisting venvs if they exist and account for platform differences in path name. This might actually be a small nice to have.

**Comment by BeastLe9enD** [comment]
> Usually, activating the venv before building your project should be sufficient as we normally use the first `python3` binary found on the path to determine the interpreter configuration.

@adamreichold I tried running `activate.ps1` (I'm on windows, so its powershell), and after that my cargo run. It does not work. when I just run the script via python, it works. Can you clarify how to reproduce?
